src/condor_io/condor_secman.cpp

```diff
@@ -903,12 +899,9 @@ SecMan::ReconcileSecurityPolicyAds(const ClassAd &cli_ad, const ClassAd &srv_ad)
 		action_ad->Assign(ATTR_SEC_AUTHENTICATION_METHODS_LIST, the_methods);
 
 		// send the single method for pre 6.5.0
-		StringList  tmpmethodlist( the_methods.c_str() );
-		char* first;
-		tmpmethodlist.rewind();
-		first = tmpmethodlist.next();
-		if (first) {
+		for (auto& first : StringTokenIterator(the_methods)) {
 			action_ad->Assign(ATTR_SEC_AUTHENTICATION_METHODS, first);
+			break;
```

Comment:
I think we can replace this loop-with-break with

```
action_ad->Assign(ATTR_SEC_AUTHENTICATION_METHODS, *StringTokenIterator(the_methods).begin());
```

Reply:
That could crash if the list is empty. Something will have gone seriously wrong if it is empty, but I can't rule it out (especially with a malicious peer).